OpenSim/Common/Component.cpp

```diff
@@ -202,19 +208,25 @@ void Component::connect(Component &root)
 
 
 // Call connect on all components and find unconnected Connectors a
```

Comment:
What are valid names? including "/" in a name would mess us up as far as I can tell, don't know about ".", "..", " ", "," etc. if used in component name. Would be good to have some error checks, generation of valid names in a central location.